source/core/serial_solid.cc

```diff
@@ -484,77 +520,57 @@ SerialSolid<dim, spacedim>::write_output_results(
 
 template <int dim, int spacedim>
 void
-SerialSolid<dim, spacedim>::write_checkpoint(std::string /*prefix*/)
+SerialSolid<dim, spacedim>::write_checkpoint(std::string prefix)
 {
-  // SolutionTransfer<dim, Vector<double>, spacedim> system_trans_vectors(
-  //   this->displacement_dh);
-
-  // std::vector<const Vector<double> *> sol_set_transfer;
-  // sol_set_transfer.push_back(&displacement);
-
-  // system_trans_vectors.prepare_for_serialization(sol_set_transfer);
+  // Checkpoint the DOF Handler
+  {
+    std::string file_name =
+      prefix + ".solid_object." + Utilities::int_to_string(id, 2) + ".dof";
+    std::ofstream                 ofs(file_name);
+    boost::archive::text_oarchive oa(ofs);
+    displacement_dh.save(oa, 0);
+  }
+  // Checkpoint the displacement vector which we will use to shift the
+  // triangulation
+  {
+    std::string file_name = prefix + ".solid_object." +
+                            Utilities::int_to_string(id, 2) + ".displacement";
+    std::ofstream                 ofs(file_name);
+    boost::archive::text_oarchive oa(ofs);
+    displacement.save(oa, 0);
+  }
 
-  // if (auto tria = dynamic_cast<parallel::distributed::Triangulation<dim> *>(
-  //       this->solid_tria.get()))
-  //   {
-  //     std::string triangulationName = prefix + ".triangulation";
-  //     tria->save(prefix + ".triangulation");
-  //   }
+  // Re-read pvd handler from output files
+  pvdhandler.save(prefix + "_solid_object_" + Utilities::int_to_string(id, 2));
 }
 
 template <int dim, int spacedim>
 void
-SerialSolid<dim, spacedim>::read_checkpoint(std::string /*prefix*/)
+SerialSolid<dim, spacedim>::read_checkpoint(std::string prefix)
 {
-  throw(std::runtime_error(
-    "Restarting DEM simulations with floating meshes is currently not possible."));
```

Comment:
Maybe add a "### Fixed" in the Changelog ?

Reply:
Will do so :)